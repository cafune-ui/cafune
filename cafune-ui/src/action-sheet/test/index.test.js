import ActionSheet from '../';
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

const mockProps = {
  title: 'test title',
  horizon: {
    wrap: false,
    list: [
      {
        name: 'hi',
        subName: 'hi sub',
        icon:
          'https://image.uisdc.com/wp-content/uploads/2018/07/uiii-ps-icon.png',
        badge: true,
        action() {
          console.log('ps hi');
        }
      },
      {
        name: 'hi2',
        subName: 'hi2 sub',
        icon: 'search',
        badge: '2',
        action: '/'
      },
      {
        name: 'hi2',
        subName: 'hi2 sub',
        icon: 'search',
        badge: '2'
      },
      {
        name: 'hi2',
        subName: 'hi2 sub',
        icon: 'search',
        badge: '2'
      },
      {
        name: 'hi2',
        subName: 'hi2 sub',
        icon: 'search',
        badge: '2'
      }
    ]
  },
  vertialList: [
    {
      name: 'browser',
      icon: 'browser'
    },
    {
      name: 'error',
      subName: 'on disabled state',
      icon: 'error',
      align: 'center',
      isDisabled: true
    },
    {
      name: 'refresh',
      subName: 'on loading state',
      icon: 'refresh',
      align: 'right',
      isLoading: true
    }
  ],
  cancelText: 'cancel',
  confirmText: 'confirm'
};
describe('ActionSheet', () => {
  it('should render properly', () => {
    const onCancel = jest.fn();
    const wrapper = shallow(
      <ActionSheet isShow {...mockProps} onClose={onCancel} showConfirm />
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.render(<ActionSheet isShow={false} {...mockProps} onClose={onCancel} />);
    setTimeout(() => {
      expect(onCancel).toHaveBeenCalled();
    }, 300);
  });

  it('should handle action', () => {
    const onConfirm = jest.fn();
    const onClose = jest.fn();
    const onOpen = jest.fn();
    const wrapper = deep(
      <ActionSheet isShow {...mockProps} onOpen={onOpen} onClose={onClose} showConfirm showCancel onConfirm={onConfirm} closeOnClickMask />
    );
    expect(wrapper).toMatchSnapshot();

    setTimeout(() => {
      expect(onOpen).toHaveBeenCalled();
    }, 300);
    wrapper.find('.caf-actsheet__action__btn--confirm').simulate('click');
    expect(onConfirm).toHaveBeenCalled();
    wrapper.find('.caf-actsheet__action__btn--cancel').simulate('click');
    expect(onClose).toHaveBeenCalledTimes(2);

    wrapper.find('.caf-actsheet__wrapper').simulate('click', {
      target: {
        className: 'caf-actsheet__wrapper'
      }
    });
    expect(onClose).toHaveBeenCalledTimes(3);
  });
});
