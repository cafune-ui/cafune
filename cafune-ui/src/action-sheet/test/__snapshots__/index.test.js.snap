// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet should handle action 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "cancelText": "cancel",
      "closeOnClickMask": true,
      "confirmText": "confirm",
      "horizon": Object {
        "list": Array [
          Object {
            "action": [Function],
            "badge": true,
            "icon": "https://image.uisdc.com/wp-content/uploads/2018/07/uiii-ps-icon.png",
            "name": "hi",
            "subName": "hi sub",
          },
          Object {
            "action": "/",
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
        ],
        "wrap": false,
      },
      "isShow": true,
      "onClose": [MockFunction],
      "onConfirm": [MockFunction],
      "onOpen": [MockFunction],
      "showCancel": true,
      "showConfirm": true,
      "title": "test title",
      "vertialList": Array [
        Object {
          "align": "left",
          "icon": "browser",
          "name": "browser",
        },
        Object {
          "align": "center",
          "icon": "error",
          "isDisabled": true,
          "name": "error",
          "subName": "on disabled state",
        },
        Object {
          "align": "right",
          "icon": "refresh",
          "isLoading": true,
          "name": "refresh",
          "subName": "on loading state",
        },
      ],
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`ActionSheet should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "cancelText": "cancel",
      "confirmText": "confirm",
      "horizon": Object {
        "list": Array [
          Object {
            "action": [Function],
            "badge": true,
            "icon": "https://image.uisdc.com/wp-content/uploads/2018/07/uiii-ps-icon.png",
            "name": "hi",
            "subName": "hi sub",
          },
          Object {
            "action": "/",
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
          Object {
            "badge": "2",
            "icon": "search",
            "name": "hi2",
            "subName": "hi2 sub",
          },
        ],
        "wrap": false,
      },
      "isShow": true,
      "onClose": [MockFunction],
      "showConfirm": true,
      "title": "test title",
      "vertialList": Array [
        Object {
          "align": "left",
          "icon": "browser",
          "name": "browser",
        },
        Object {
          "align": "center",
          "icon": "error",
          "isDisabled": true,
          "name": "error",
          "subName": "on disabled state",
        },
        Object {
          "align": "right",
          "icon": "refresh",
          "isLoading": true,
          "name": "refresh",
          "subName": "on loading state",
        },
      ],
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
